src/duk_bi_global.c

```diff
@@ -980,8 +981,24 @@ DUK_LOCAL void duk__bi_global_resolve_module_id(duk_context *ctx, const char *re
 		}
 	}
  loop_done:
-
+	/* Output #1: resolved absolute name */
 	duk_push_lstring(ctx, (const char *) buf_out, (size_t) (q - buf_out));
+
+	/* Get last component of resolved ID: backtrack to previous slash or
+	 * start of buffer.
+	 */
+	q_last = q;
+	while (q_last > buf_out && *(q_last - 1) != '/') {
```

Comment:
Is the backtrack scan necessary?  The ID string was already scanned once, presumably the resolver could just note the position of the final slash and then we'd use that here.